source/deployment-guide/mobile/configure-microsoft-intune-mam.rst

```diff
@@ -0,0 +1,398 @@
+* Your identity strategy cannot use Azure AD ``objectId`` as the authoritative user identifier.
+* You need a rollout model where users can defer or bypass Intune enrollment.
+
+Before You Continue
+-------------------
+
+Before proceeding, confirm the following are true:
+
+* You use Microsoft Entra ID for authentication.
+* You can commit to Azure AD ``objectId`` as the authoritative identity.
+* You have (or can obtain) a Mattermost Enterprise Advanced license.
+* Target users are licensed for Microsoft Intune.
+* You can register applications and grant admin consent in Microsoft Entra.
+
+If any of the above are not true, do not proceed.
+
+Configuration Overview
+----------------------
+
+Configuring Intune MAM for Mattermost Mobile requires coordinated setup across the following four systems:
+
+* **Microsoft Entra ID (Azure AD)** – identity, app registration, API permissions
+* **Microsoft Intune** – app protection policies and user targeting
+* **Mattermost Server** – MAM enablement and identity alignment
+* **Mattermost Mobile App (iOS)** – enrollment and enforcement
+
+If any system is misconfigured, Intune MAM enrollment will fail.
+
+Before beginning configuration, review the `Identity Configuration <#identity-configuration-for-intune-mam>`__ section to confirm your deployment meets the required identity model.
+
+Setup Order
+~~~~~~~~~~~
+
+Follow this setup order exactly to avoid enrollment failures and rework.
+
+Step 1: Finalize Identity Strategy (Entra + Mattermost)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Commit to Azure AD ``objectId`` as the authoritative identity.
+* Ensure all authentication methods (OAuth, SAML, LDAP) resolve to the same value.
+* Confirm access tokens include the ``oid`` claim.
+
+Identity must be finalized before enabling Intune or enrolling users.
+
+Step 2: Configure Microsoft Entra Applications & Permissions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Register the Mattermost mobile app as a public/native Entra application.
+* Copy the **Application (Client) ID**.
+
+  To register an application in Microsoft Entra, see: https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app
+
+* Configure the iOS platform with the correct bundle ID and redirect URI.
+
+  For Entra portal steps, see: https://learn.microsoft.com/en-us/entra/identity-platform/scenario-mobile-app-configuration
+
+  For redirect URI formatting details, see: https://learn.microsoft.com/en-us/entra/identity-platform/how-to-add-redirect-uri
+
+* Grant required Intune MAM API permissions with admin consent.
+
+  To grant tenant-wide admin consent, see: https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/grant-admin-consent
+
+Step 3: Configure Mattermost Server for Intune MAM
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Enable Intune MAM in the System Console.
+* Set ``IdAttribute = objectId``.
+* Verify Enterprise Advanced licensing.
```

Comment:
Here they will need to add the `Application (client) ID` and the `Directory (tenant) ID` of the Application they just registered in Step 2 as well as Selecting their Auth Provider (available based on what is configured OpenID Connect with Office 365 or SAML).

Important: For SAML we are not really validating that the IdP is Entra, so perhaps we must make absolutely sure they understand that to use SAML they must use it with Entra ID (I know is in Before you Continue, but we can mention it here too)